### PR TITLE
Change docker-compose to support ARM

### PR DESCRIPTION
**Pull request by istareatscreens**
**Description of issue**
- The current app docker image does not support ARM processors. It would be nice to be able to run this server on a Raspberry pi/mac
- Furthermore it would be nice to be able to specify database access outside of the local network (for instances when you are connecting to the server via LAN). At the moment the port is mapped to 127.0.0.1 so it cannot be accessed outside of the host.
- It would be good to add a failure condition to the app just in case it fails during initial build, I found this sometimes occurred during migration and just restarting the container resolved the issue.
- Using volume instead of specifying a local path for volume mapping can lead to data loss (due to deletion of volumes)

**Changes:**
- I switched the repo image with a Dockerfile that supports ARM and x64 processors
- Added ```"${AQUADX_DB_PORT:-127.0.0.1:3369}:3306"``` to the docker file for the db port mapping so that non-local can be specified without modifying docker-compose so you can start the server conveniently with: ```AQUADX_DB_PORT=3369 docker compose up``` to get a custom 0.0.0.0 port map for access over LAN
- Add restart on fail 3 times to handle any startup failures due to initial migration that may occur during app build
- Added ```"${AQUADX_DB_DATA:-mariadb_data}:/var/lib/mysql"``` to allow custom mapping to local drive over using a volume which can easily be deleted. Added a gitignore directory for storing these files e.g. ```AQUADX_DB_PORT=3369 AQUADX_DB_DATA=$PWD/db docker compose up```
- Removed depreciated 'version' in docker-compose

**Notes:**
- The server builds properly on both ARM and x64 processors
- Access over lan to database is much easier to setup
- Downsides app is slower to build initially

**Suggestion:**
- It may be wise to switch from ```mariadb:latest``` to a specific version to prevent the server breaking due to a database update.
- Update repo image to use this new Dockerfile to circumvent having to build the Dockerfile

### CODE REVIEW

**Comment by clansty** [comment]
lgtm

**Comment by Menci** [comment]
Why build the Docker image on starting? Is there any reason not to push a multi-arch image to the repo and use that?

**Comment by istareatscreens** [comment]
> Why build the Docker image on starting? Is there any reason not to push a multi-arch image to the repo and use that?

Not sure if I have the ability to push the image to the repo. Ideally someone with access could add the image to the repo then edit this PRs docker-compose file to point to the repo image of its current Dockerfile.

I included this as one of my suggestions:
image

Also if this is merged and if desired I can update the FAQ to include documentation explaining these settings.

File: docker-compose.yml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
   app:
-    image: hykilpikonna/aquadx:latest
+    build:
+      context: .
+      dockerfile: Dockerfile
     ports:
       - "80:80"
       - "8443:8443"
       - "22345:22345"
+    restart: on-failure:3
```
Comment (hykilpikonna) [inline docker-compose.yml:10]: :3

**Comment by hykilpikonna** [comment]
> Ideally someone with access could add the image to the repo then edit this PRs docker-compose file to point to the repo image of its current Dockerfile.

Right the last build was 6 mo ago... I'll try to add an action that automate this

**Comment by hykilpikonna** [comment]
Ok just realized building arm64 with qemu on x86 is probably a bad idea...

image

**Comment by hykilpikonna** [comment]
It took 23 minutes :o

image

**Comment by hykilpikonna** [comment]
Ok please test it and let me know if it works!

**Comment by istareatscreens** [comment]
Confirmed latest V1 is working on ARM (raspberry pi 4)

image

Thanks for the merge :3

**Pull request merged**